cli/awsbatch/awsbsub.py

```diff
@@ -179,67 +189,50 @@ def _upload_to_s3(boto3_factory, s3_bucket, file_path, key_name, timeout):
         expires += datetime.timedelta(seconds=timeout)
 
     s3_client = boto3_factory.get_client("s3")
-    s3_client.upload_file(file_path, s3_bucket, _prepend_s3_folder(key_name), ExtraArgs={"Expires": expires})
+    s3_client.upload_file(file_path, s3_bucket, key_name, ExtraArgs={"Expires": expires})
 
 
-def _upload_and_get_command(boto3_factory, args, job_name, region, s3_bucket, log):
+def _upload_and_get_command(boto3_factory, args, job_name, config, log):
     """
     Get command by parsing args and config.
 
-    The function will also perform an s3 upload, if needed
+    The function will also perform an s3 upload, if needed.
     :param boto3_factory: initialized Boto3ClientFactory object
     :param args: input arguments
     :param job_name: job name
-    :param region: region
-    :param s3_bucket: S3 bucket to use
+    :param config: config object
     :param log: log
     :return: command to submit
     """
-    if args.command_file or not sys.stdin.isatty():
+    if args.command_file or not sys.stdin.isatty() or args.env:
         # define job script name
-        job_script = "job-{0}-{1}.sh".format(job_name, int(time.time() * 1000))
+        job_id = "job-{0}-{1}".format(job_name, int(time.time() * 1000))
+        job_folder = _get_job_folder(job_id)
+        job_script = job_id + ".sh"
         log.info("Using command-file option or stdin. Job script name: %s" % job_script)
 
+        env_file = job_id + ".env.sh"
```

Comment:
What about leaving env_file to None when not used rather than repeating the check on args.env in the _compose_bash_command function?